pallets/clusters/src/lib.rs

```diff
@@ -22,42 +26,42 @@ mod benchmarking;
 
 /// The **settings** of a **Role**
 #[derive(Encode, Decode, Clone, PartialEq, Debug, Eq, scale_info::TypeInfo)]
-pub struct RoleSetting {
+pub struct RoleSetting<BoundedName, BoundedData> {
 	/// Name of the setting
-	pub name: Vec<u8>,
+	pub name: BoundedName,
```

Comment:
I think all those should rather be `BoundedVec<u8, TNameLimit>` vec u8 doesn't ever change.

Reply:
I tried that and I get a lots of errors due to missing codecs and implementations of macro. I spent two days on it.
I will try again and post here the errors I get.

Reply:
It's the Get you are using for the config that misses `WrapperTypeEncode`, which is fine to use, leave it like this, it's minor.
You could have simply explained this btw also no need to spend two days on such issues 🙏